yt/visualization/line_plot.py

```diff
@@ -31,6 +31,12 @@
     linear_transform, \
     invalidate_plot
 
+class LineObject(object):
+    def __init__(self, start_point, end_point, ds, label=None):
+        self.start_point = _validate_point(start_point, ds, start=True)
+        self.end_point = _validate_point(end_point, ds)
+        self.label = label
```

Comment:
Fair enough. Maybe it should be called `LineBuffer`?

@lindsayad hope you don't mind the bikeshedding :)